.github/workflows/base.yml

```diff
@@ -344,18 +363,47 @@ jobs:
           cd quarkus
           bash ../workflow-mandrel/.github/update_quarkus_version.sh 2.2.999
         fi
-    # Use Java 17 to build Quarkus as it doesn't build with Java 20
+    # Use Java 11 to build Quarkus as that's the lowest supported JDK version currently
     - uses: actions/setup-java@v3
       with:
         distribution: 'temurin'
-        java-version: '17'
+        java-version: '11'
+    - name: Download GraalVM Maven Repo
+      if: "!startsWith(matrix.os-name, 'windows')"
+      uses: actions/download-artifact@v3
+      with:
+        name: org-graalvm-artefacts-${{ needs.get-test-matrix.outputs.artifacts-suffix }}
+        path: .
+    - name: Download GraalVM Maven Version
+      if: "!startsWith(matrix.os-name, 'windows')"
+      uses: actions/download-artifact@v3
+      with:
+        name: mandrel-maven-version-${{ needs.get-test-matrix.outputs.artifacts-suffix }}
+        path: .
+    - name: Extract GraalVM Maven Repo and GraalVM Maven Version
+      if: "!startsWith(matrix.os-name, 'windows')"
+      shell: bash
+      run: |
+        tar -xzvf graalvm-maven-artefacts.tgz -C ~
+        tar -xzvf graalvm-version.tgz -C $(dirname ${MANDREL_HOME})
     - name: Build quarkus
       run: |
+        rm -f maven_graalvm_before_build.txt maven_graalvm_after_build.txt
+        find ~/.m2/repository/org/graalvm | sort > maven_graalvm_before_build.txt
+        GRAAL_MVN_ARTIFACTS_VERS=$(cat ${MANDREL_HOME}/.maven-version)
+        ls ${MANDREL_HOME}/.maven-version
+        echo "Building quarkus with locally installed GraalVM maven artefacts in version: ${GRAAL_MVN_ARTIFACTS_VERS}"
         cd ${QUARKUS_PATH}
-        ./mvnw -e -B --settings ${QUARKUS_PATH}/.github/mvn-settings.xml  -Dquickly
+        ./mvnw -e -B --settings ${QUARKUS_PATH}/.github/mvn-settings.xml -Dquickly -Dgraal-sdk.version="${GRAAL_MVN_ARTIFACTS_VERS}"
+        cd -
+        find ~/.m2/repository/org/graalvm | sort > maven_graalvm_after_build.txt
+        diff -u maven_graalvm_before_build.txt maven_graalvm_after_build.txt
```

Comment:
This is to check we have all dependencies installed, right?

Reply:
This is to check that the Quarkus build doesn't download any `org.graalvm*` maven artefacts from maven central in a version that we don't expect as that would make the build (wrongly) pass. Does that make sense?

Reply:
Yes, that's what I meant as well. All good.